wire-ios-sync-engine/Source/Use cases/DisableAnalyticsUseCase.swift

```diff
@@ -29,21 +29,35 @@ public protocol DisableAnalyticsUseCaseProtocol {
 
 /// Concrete implementation of the DisableAnalyticsUseCaseProtocol.
 /// This struct is responsible for disabling analytics sharing.
-public struct DisableAnalyticsUseCase: DisableAnalyticsUseCaseProtocol {
+struct DisableAnalyticsUseCase<UserSession>: DisableAnalyticsUseCaseProtocol
```

Comment:
I don't think you need generics here. In the initializer you can just do:

`userSession: DisableAnalyticsUseCaseUserSession`

Reply:
I guess you are using Generics to have static vs dynamic dispatch like @caldrian in some of his PRs. Is that correct?

Reply:
Removed the generics from the use cases: https://github.com/wireapp/wire-ios/commit/c82f5d3b1e3b7847902aa182978145fcd4e241d5